xdp-dump/README.org

```diff
@@ -0,0 +1,126 @@
+#+EXPORT_FILE_NAME: xdpdump
+# This file serves both as a README on github, and as the source for the man
+# page; the latter through the org-mode man page export support.
+# .
+# To export the man page, simply use the org-mode exporter; (require 'ox-man) if
+# it's not available.
+# .
+# The org-mode export doesn't support extended title lines, so manually copy
+# over the first line of the resulting .man file before exporting and
+# committing.
+
+* xdpdump - a simple tcpdump like tool for capturing packets at the XDP layer
+=xdpdump= is a simple XDP packet capture tool that tries to behave similar to
+=tcpdump=, however, it has no packet filter or decode capabilities.
+
+=xdpdump= works by attaching a bpf trace program to the XDP entry and/or exit
+function which stores the raw packet in a perf trace buffer. If no XDP program
+is loaded this approach can not be used and the tool will use a libpcap
+live-capture to be backward compatible.
+
+
+** Running xdpdump
+The syntax for running =xdpdump= is:
+
+#+begin_src
+Usage: xdpdump [options]
+
+ XDPDump tool to dump network traffic
+
+Options:
+     --rx-capture <mode>    Capture point for the rx direction (valid values: fentry,fexit)
+ -D, --list-interfaces      Print the list of available interfaces
+ -i, --interface <ifname>   Name of interface to capture on
+ -s, --snapshot-length <snaplen>  Minimum bytes of packet to capture
+ -w, --write <file>         Write raw packets to pcap file
+ -v, --verbose              Enable verbose logging (-vv: more verbose)
+ -h, --help                 Show this help
+#+end_src
+
+* The options explained
+The =xdpdump= tool tries to mimic the basic =tcpdump= options, but just in case
+below each of the available options is explained:
+
+** --rx-capture <mode>
+Specify where the ingress packet gets captured. Either at the entry of the XDP
+program and/or exit of the XDP program. Valid options are *fentry*, *fexit*,
+or both *fentry,fexit*. The packet at *fexit* can be modified by the XDP
+program. If you are interested to see both the original and modified packet,
+use the *fentryfexit* option. With this, each packet is captured twice.
```

Comment:
Missing comma between fentry and fexit in this last line. Also, maybe specify that what 'fentry' and 'fexit' means? :)

Reply:
`-use the *fentryfexit* option. With this, each packet is captured twice.
+use the *fentry,fexit* option. With this, each packet is captured twice.
+fentry and fexit refer to the kernels function tracer framework, i.e. function
+entry or exit. In the XDP context, it means the XDP program's entry and exit
+points.
`

Reply:
Hmm, now that you updated this, I'm wondering whether it wouldn't be less confusing to just drop the 'f'? I.e., just name the user-visible flags 'entry' and 'exit' and drop everything except the last bit of that help message you just added? :)

Reply:
Will change it to 

> Specify where the ingress packet gets captured. Either at the entry of the XDP
program and/or exit of the XDP program. Valid options are *entry*, *exit*,
or both *entry,exit*. The packet at *exit* can be modified by the XDP
program. If you are interested to see both the original and modified packet,
use the *entry,exit* option. With this, each packet is captured twice. In the
XDP context, it means the XDP program's entry and exit points.

Reply:
With that change I don't even think you need the last sentence :)